### DISCUSSION TITLE
Recommended linter for contributing

### DISCUSSION THREAD

**Discussion by Mathieu4141** [ContinualAI/avalanche#630] [Q&A] [answered]
Hi all! 🙂 

<details> <summary>I'm a big fan! 🚀 </summary>
First I must say that I'm really impressed with avalanche, I really like how it is intuitive and easy to use. And I'm definitively a big fan of the plugin system, being able to combine different methods is super great!
We've used avalanche for a school project, so I'm pushing a few things that we wrote and I think could benefit everyone using the lib 🙂 
</details> 

Is there a recommended linter / formatter to use when contributing to avalanche? So far I used PyCharm "Reformat code" tool, but it doesn't seem to match the formatting you use, and I end up manually removing formatting changes that would pollute the PRs

**Comment by AndreaCossu** [comment, marked as answer]
@Mathieu4141 that's the kind of interest we really like to see, so thank you for interacting :smile: Regarding the code formatting, we simply stick to the pep8 style. If you run `pycodestyle avalanche` and `pycodestyle examples` in the project root and no errors comes up you are fine :smile: Pycharm should warn you if you are not pep8 compliant by default.
